src/connection.ts

```diff
@@ -60,14 +61,22 @@ export class WebRTCConnection implements ic.Connection {
       },
     };
     this.handleIncomingStreams();
+    this.peerConnection.onconnectionstatechange = (_) => {
```

Comment:
why not just put `const closed = ['closed', 'failed', 'disconnected'].includes(this.peer.connectionState);`

Avoids the whole switch statement etc.